### PR TITLE
Add selenium tests to travis CI

### PR DESCRIPTION
**Pull request by imalsogreg**
After many fake commits bumping travis, I finally landed on a solution where Travis runs selenium tests.

Main change: `.travis.yml` script makes a call to `nix build -f travis.nix testresults`. The `testresults` attribute depends on servant-reflex example frontend, the example testserver, and the selenium webdriver executable.

There used to be several ad-hoc scripts for integrating with `try-reflex`, `work-on`, bash copying js outputs into testserver static directory... all of that is replaced by nix code and organized nicely (I hope!) within `travis.nix`.

### CODE REVIEW

**Comment by imalsogreg** [comment]
@mightybyte @3noch @kmicklas @srhb   This is a fairly big change to how I advise people to work with servant-reflex, cargo-culting on top of Sarah's travis-ci PR. Would any of you mind giving it a look-over to see that it's reasonable?

File: nix/testresults.nix [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  echo About to phantom
+  ${phantomjs}/bin/phantomjs --webdriver=127.0.0.1:4444 &
+  sleep 3
+  echo About to Server
+  STATIC_DIR=${testserver}/static ${testserver}/back -q --no-access-log --no-error-log -p 8000 &
+  sleep 3
+  ${curl}/bin/curl localhost:8000
+  ${curl}/bin/curl localhost:8000/runmain.js
+  echo About to testdrive
+  ${testdriver}/bin/spec
```
Comment (srhb) [inline nix/testresults.nix]: From the name `testresults` I would expect `${testdriver}/bin/spec > $out 2>&1` or something similar

Comment (imalsogreg) [reply nix/testresults.nix]: Great. The actual test results are much more informative than `Done`! :)

```
[greghale@p51:~/code/servant-reflex]$ cat result

Servant.Reflex
  servant-reflex tests
    test page using chrome
      opens the page
      has a div
      has demo-group div
      doesn't have demo-group-nonexistent div
      can get an int or unit from the API
      handle query params
      handles queryflag
      tags dyn promptly

Finished in 3.1099 seconds
8 examples, 0 failures
```

**Comment by imalsogreg** [comment]
@srhb Thanks a million!

**Pull request merged**